pysnooper/tracer.py

```diff
@@ -272,27 +278,23 @@ def trace(self, frame, event, arg):
 
         ### Reporting newish and modified variables: ##########################
         #                                                                     #
-        old_local_reprs = self.frame_to_local_reprs[frame]
+        old_local_reprs = self.frame_to_local_reprs.get(frame, {})
         self.frame_to_local_reprs[frame] = local_reprs = \
                                        get_local_reprs(frame, watch=self.watch)
 
-        modified_local_reprs = {}
-        newish_local_reprs = {}
-
-        for key, value in local_reprs.items():
-            if key not in old_local_reprs:
-                newish_local_reprs[key] = value
-            elif old_local_reprs[key] != value:
-                modified_local_reprs[key] = value
-
         newish_string = ('Starting var:.. ' if event == 'call' else
                                                             'New var:....... ')
-        for name, value_repr in sorted(newish_local_reprs.items()):
-            self.write('{indent}{newish_string}{name} = {value_repr}'.format(
-                                                                   **locals()))
-        for name, value_repr in sorted(modified_local_reprs.items()):
-            self.write('{indent}Modified var:.. {name} = {value_repr}'.format(
+
+        for name, value_repr in local_reprs.items():
+            if name not in old_local_reprs:
+                self.write('{indent}{newish_string}{name} = {value_repr}'.format(
+                                                                       **locals()))
+
+        for name, value_repr in local_reprs.items():
+            if name in old_local_reprs and old_local_reprs[name] != value_repr:
+                self.write('{indent}Modified var:.. {name} = {value_repr}'.format(
```

Comment:
This is actually not what I had in mind. I guess you're trying to make sure that new variables are all written before modified ones, which I hadn't thought about. I'm not actually sure if that's a constraint worth sticking to. Maybe variable declaration order should always win?

Reply:
I don't mind either way. Alex, feel free to make a decision here.

Reply:
OK, let's make variable declaration order king. Flatten the code into a single loop.

Reply:
Yes, the variable declaration order still wins, because the first message is always about the declaration and the further ones - about modifying.
The only exception I know if when we use watch. For example if we have a list of [7,8,9] and append 10 to it we can get:

    Modified var:.. lst = [7, 8, 9, 10] 
    New var:....... (lst + [])[3] = 10
    Modified var:.. lst + [] = [7, 8, 9, 10]

I added a test on ordering and another test on cell variables in nested functions.

Reply:
Are you saying that sample output is what you currently get, or what you would get if you changed the code? Is it good or bad?

Reply:
Currently the new vars are always before the modified ones. This is corresponds to the old behavior (before this pull request).

For me the both variants seems good.

Reply:
Right, I'm saying let's drop that old behaviour. New and modified can be in any order, I don't think it improves anyone's experience, @cool-RR seems to be saying it's fine. That's what I mean by variable declaration order is king, it's the only consideration, and I think that's good for both the code and the user.

Reply:
Ok, done, https://github.com/cool-RR/PySnooper/pull/113/commits/21d59bc6120f6072ac3656e16e55948424e3a1b3